src/Connection/Connection.php

```diff
@@ -12,25 +12,26 @@ interface Connection
 {
     public const MAX_KEEP_ALIVE_TIMEOUT = 60;
 
+    /**
+     * @return Promise<Stream>
+     */
+    public function getStream(): Promise;
```

Comment:
Do we really need this separate method?

Reply:
The purpose was that the initial request for a stream from a connection should be independent of the request and never resolve with null.

Is that really necessary? Probably not, the connection can know if it's the first time `getStreamFor()` is called and never fail.